src/core/array.c

```diff
@@ -288,11 +285,11 @@ JANET_CORE_FN(cfun_array_insert,
     if (at < 0) {
         at = array->count + at + 1;
     }
-    if (at < 0 || at > array->count)
+    if (at < 0 || (size_t) at > array->count)
```

Comment:
So now how do we access elements in the middle of a large array? The cast indicates a logic issue.

Reply:
Now the code is using a signed size type in the `at` variable. Since the first part of the short-circuit logic is that `at < 0`, then, we can cast it to `size_t` when it is between `[0,JANET_INTMAX_SIZE]`. Is that a good solution?

Reply:
LGTM, as long as ssize_t is stubbed out when not defined.